src/fit.jl

```diff
@@ -3,56 +3,62 @@
 #    ACE Fitting
 
 import IPFitting, ACE1pack, ACE1
+using Dates, Base 
 
-export fit_params, fit_ace 
+export fit_params, fit_ace, make_ace_db, db_params, fit_ace_db, save_fit
 
 """
-TODO: documentation:
+`fit_ace(params::Dict)` : function to set up and fit the least-squares 
+problem of "atoms' positions" -> "energy, forces, (virials)". Takes in a 
+dictionary with all the parameters. see `?fit_params` for details. 
+
+Returns `IP`, `lsqinfo`
 """
 function fit_ace(params::Dict)
 
-    # ENH: don't generate redundant stuff if db is read from disk. 
-    # ENH: make species non-mandatory and read from data
-    data =  ACE1pack.read_data(params["data"])
-
-    basis = [ACE1pack.generate_basis(basis_params) for (basis_name, basis_params) in params["basis"]]
-    basis = JuLIP.MLIPs.IPSuperBasis(basis);
-
-    if params["fit_from_LSQ_DB"]
-        db = LsqDB(params["LSQ_DB_fname_stem"])
-    else
-        db = LsqDB(params["LSQ_DB_fname_stem"], basis, data)
-    end
-
-
-    solver = ACE1pack.generate_solver(params["solver"])
-
-    if !isnothing(params["P"]) 
-        solver["P"] = ACE1pack.generate_precon(basis, params["P"])
-    end
-
-    if typeof(params["e0"]) == Dict{Any, Any}
-        # sometimes gets read in (from yaml?) as Dict{Any, Any} 
-        # which gives StackOverflowError somewhere in OneBody
-        params["e0"] = convert(Dict{String, Any}, params["e0"])
-    end
-
-    Vref = OneBody(params["e0"])
-
-    weights = params["weights"]
-
-    IP, lsqinfo = lsqfit(db, solver=solver, Vref=Vref, weights=weights, error_table=true)
-
-    @info("Fitting errors")
-    rmse_table(lsqinfo["errors"])
-
-    lsqinfo["fit_params"] = params 
-
-    _save_fit(params["ACE_fname"], IP, lsqinfo)
+    db = _decide_how_to_get_db(params)
+    IP, lsqinfo = fit_ace_db(db, params)
 
     return IP, lsqinfo
 end
 
+
+"""
+`fit_params(; kwargs...)` : returns a dictionary containing all of the
+parameters needed for making an ACE potential. All parameters are passed 
+as keyword argumts. 
+
+### Parameters
+* `data` : data parameters, see `?data_params` for details (mandatory)
+* `basis` : dictionary containing dictionaries that specify the basis used
+in fitting. Usually just `rpi_params` and `pair_params` (mandatory). 
+For example
+```
+basis = Dict{
+    "rpi" => rpi_params(; kwargs...),
```

Comment:
I wonder whether we should call this `"ace"` instead of `"rpi"`. Or `"mb"`

Reply:
I definitely have no idea what `rpi` might stand for

Reply:
rotation and permutation invariant :)

Reply:
Note there is also a type here it should be `Dict(` not `Dict{`

Reply:
yes, that sounds good. I think arbitrary labels was the intention initially, but I have likely messed that up. I'll add these suggestions. Except maybe for "basistype", I'd rather have "type", as for other entries with choices. I think the context should be clear